tools/api_proto_breaking_change_detector/buf_utils.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+from typing import List, Union, Tuple
+
+from detector_errors import ChangeDetectorError, ChangeDetectorInitializeError
+from tools.base.utils import cd_and_return
+from tools.run_command import run_command
+
+
+def _generate_buf_args(target_path, config_file_loc, additional_args):
+    buf_args = []
+
+    # buf requires relative pathing for roots
+    target_relative = Path(target_path).absolute().relative_to(Path.cwd().absolute())
+
+    # buf does not accept . as a root; if we are already in the target dir, no need for a --path arg
+    if str(target_relative) != ".":
+        buf_args.extend(["--path", str(target_relative)])
+
+    if config_file_loc:
+        buf_args.extend(["--config", str(config_file_loc)])
+
+    buf_args.extend(additional_args or [])
+
+    return buf_args
+
+
+def _cd_into_config_parent(func):
+
+    def cd_wrapper(*args, **kwargs):
+        config_file_loc = kwargs.get("config_file_loc")
+        config_parent = Path(config_file_loc).parent if config_file_loc else Path.cwd()
+        with cd_and_return(config_parent):
+            return func(*args, **kwargs)
+
+    return cd_wrapper
```

Comment:
This is really nice and fancy, but it obscures the fact that the `cd` is happening, and relies on some magic (specifically the caller having a `config_file_loc` argument). I think a cleaner solution would be to have this just return the result of cd_and_return, which is a context and then use `with` in the wrapped functions, something like this:
```python
def pull_buf_deps(...):
    with _cd_into_config_parent(config_file_loc):
        # run_command, etc.
```